src/Struct/FindologicProduct.php

```diff
@@ -762,14 +762,30 @@ protected function setOrdernumberByProduct(ProductEntity $product): void
     protected function setCategoriesAndCatUrls(): void
     {
         $productCategories = $this->product->getCategories();
-        if ($productCategories === null || empty($productCategories->count())) {
+        $productVariants = $this->product->getChildren();
+
+        // If there are no product categories, and no children either, we throw ProductHasNoCategoriesException.
+        if ($productCategories->count() === 0 && $productVariants->count() === 0) {
             throw new ProductHasNoCategoriesException($this->product);
         }
 
         $catUrls = [];
         $categories = [];
 
         $this->parseCategoryAttributes($productCategories->getElements(), $catUrls, $categories);
+
+        if ($productVariants) {
+            foreach ($productVariants as $variant) {
+                $variantCategories = $variant->getCategories();
+                if (!$variantCategories) {
```

Comment:
This is similar to the check in line 777. Consider using the `count()` method of the collection.